index.html

```diff
@@ -810,11 +852,8 @@ <h2>Self-description</h2>
                 <dt>HTTP</dt>
                 <dd>
                     <p>
-                        <span class="rfc2119-assertion" id="self-http-secure"> 
-                            The HTTP-based self-description SHOULD be over HTTPS (HTTP Over TLS).
-                        </span>
                         <span class="rfc2119-assertion" id="self-http-access-control">
-                            The server SHOULD serve the requests after performing necessary 
+                            The server MUST serve the requests only after performing necessary 
```

Comment:
I'm concerned about this side-effect. I'm reading this as if every web thing is not allowed to publicly expose its TD, is that intended? Which are the downsides of allowing the possibility of exposing a TD to the public but allowing authenticated access for all (or a subset) of its affordances?

Reply:
I agree it should be possible to have publicly accessible TDs which require no authentication/authorisation. I think this assertion does allow that by using the word _necessary_, but perhaps the language could be made clearer.

Another problem with this assertion though is that technically serving a 401 Unauthorized response is serving a request before authentication/authorization has taken place. The assertions might make sense if split up by different scenarios, e.g.

- If not authenticated where authentication is required, MUST respond with 401
- If authenticated but not authorized, MUST respond with 403 (maybe 404 in some cases?)
- If authenticated and authorized (where required), MUST respond with 200
- If the TD is not found, MUST respond with 404

Reply:
My point here is that you if say "SHOULD" and "necessary" together it's an oxymoron.  It would allow people to do the necessary authentication and authorization checks, find they fail, and STILL serve the TD, which is obviously not what we want.  If authentication and authorization fail, the resource should not be served, full stop.

Reply:
I'll have to think about the different scenarios.  Basically if authentication/authorization fails we don't want to serve the *resources*, but might still *reply* (with an error).  Let me think if I can tweak the wording to cover that.

Reply:
> My point here is that you if say "SHOULD" and "necessary" together it's an oxymoron. It would allow people to do the necessary authentication and authorization checks, find they fail, and STILL serve the TD, which is obviously not what we want. If authentication and authorization fail, the resource should not be served, full stop. If you want to serve a TD publically, then you can use nosec... IF it is permitted (does not have PII). In that case the "necessary" checks will always return "true" and the MUST condition is satisfied. So the assertion still allows for "public" TD distribution using nosec, as long as it's not PII.

Ah, I see your point, but I still would ask you to improve the sentence. I asked around a bunch of colleagues and all of them got the same impression. MUST together with "necessary" implies that you are always going to check for authentication and authorization and "no-check" feels that is not an option. I would like to see your comment above as an explainer to that assertion, maybe pointing to the fact that in WoT no_sec is still an authentication/authorization policy. Or maybe change the sentence to: 
>  The server MUST respond with the requested TD only after performing authentication and authorization if necessary.

Reply:
> Regarding the ordering issue brought up by Ben I'll have to think about the different scenarios. Basically if authentication/authorization fails we don't want to serve the resources, but might still reply (with an error). Let me think if I can tweak the wording to cover that.

I would also agree that this is an issue. Explaining the different scenarios like @benfrancis is suggesting might also clarify the assertion too.